### ISSUE TITLE
wildcard not work under query_string

### ISSUE THREAD

**Issue by tm2018** [Yelp/elastalert#1719]
my query rule like this:
- query:
    query_string:
      query: "NOT geoip.ip:myip"
- query:
    query_string:
      analyze_wildcard: true
      query: "!geoip.ip:aaa.bbb.ccc.*"

as i have servel ip which change in a range(aaa.bbb.ccc.*), i want to use wildcard,but it donnot work!
i try next rules,all of there failed!!:
###rule 1
- query
    query_string:
      analyze_wildcard: true
      query: "!geoip.ip:aaa.bbb.ccc.\\.*" 
###rule 2
- query:
     query_string:
        analyze_wildcard: true
       query: "!geoip.ip:aaa.bbb.ccc.\.*"
###rule 3
- query:
    query_string:
      analyze_wildcard: true
      query: "!geoip.ip:aaa.bbb.ccc.\.*"
###rule 4
- not:
   wildcard:
      geoip.ip: "aaa.bbb.ccc.*" ##this rule occour the error "'no [query] registered for [not]"

how should i config? please help me,thanks

**Comment by Qmando** [comment]
Is this field analyzed? That query looks correct, assuming the default analyzer is the same. Maybe you can try
``NOT geoip.ip: /aaa.bbb.ccc.*/``

You can use ``--es_debug_trace trace.log`` to see the queries being made to help debug it.

**Comment by tm2018** [comment]
yes, the field is analyzed! if i set a exact ip for example 1.1.1.1,alert will ingore the ip,but when i set ip with wildcard,the ip who matched the wildcard sent to my email still,i use --es_debug_trace trace.log to debug,i found the es_query is like this:
curl -H 'Content-Type: application/json'  -XGET 'http://localhost:9200/logstash-apache_access2018-05*/_search?pretty&ignore_unavailable=true&size=0' -d '{
  "aggs": {
    "counts": {
      "terms": {
        "field": "geoip.ip",
        "size": 50
      }
    }
  },
  "query": {
    "bool": {
      "filter": {
        "bool": {
          "must": [
            {
              "range": {
                "@timestamp": {
                  "gt": "2018-05-30T01:35:19.668263Z",
                  "lte": "2018-05-30T01:36:09.599633Z"
                }
              }
            },
            {
              "query_string": {
                "query": "*"
              }
            },
            {
              "query_string": {
                "query": "NOT geoip.ip:1.2.3.4"
              }
            },
            {
              "query_string": {
                "query": "NOT geoip.ip:wildcard: /a.b.c.*/"
              }
            }
          ]
        }
      }
    }
  }
}'
but it donnot working! maybe it is the es grammar error for my query,so i will try to put a issue in es

**Comment by Qmando** [comment]
Use ``NOT geoip.ip: /aaa.bbb.ccc.*/`` like I said, not ``geoip.ip:wildcard:``

**Comment by tm2018** [comment]
yeah,i tried NOT geoip.ip: /aaa.bbb.ccc.*/ in my rule file,i do not work still!  and i user curl to query from es with "query": "NOT geoip.ip: /aaa.bbb.ccc.*/",but the result it return to me include aaa.bbb.ccc.* 
my es version is 6.2.4,i install with pip install "elasticsearch>=5.0.0", is it the version provlem or others?i remember there is a config to edit but i forget where it is

**Comment by Qmando** [comment]
Maybe the default analyzer changed in ES6 and it splits on periods now. You could try ``geoip.ip.keyword``
